ssl/tls13_enc.c

```diff
@@ -511,6 +541,16 @@ int tls13_change_cipher_state(SSL_CONNECTION *s, int which)
                 goto err;
             }
 
+            if (!(EVP_CIPHER_flags(cipher) & EVP_CIPH_FLAG_AEAD_CIPHER)) {
+                /* Cipher is not AEAD */
+                if (!ssl_cipher_get_evp_md_mac(sctx, sslcipher, &mac_md,
+                                                &mac_pkey_type,
+                                                &mac_secret_size)){
+                    EVP_MD_CTX_free(mdctx);
```

Comment:
Maybe do the `EVP_MD_CTX_free(mdctx)` centrally at the end, after `err:`?
(Which will require moving the variable `mdctx` to top level in the function.)